asm/istio/istio-operator.yaml

```diff
@@ -11,16 +11,56 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
+#
 apiVersion: install.istio.io/v1alpha1
 kind: IstioOperator
 spec:
-  profile: asm-gcp
+  profile: empty
   hub: gcr.io/gke-release/asm # {"$ref":"#/definitions/io.k8s.cli.setters.anthos.servicemesh.hub"}
-  tag: ASM_IMAGE_TAG # {"$ref":"#/definitions/io.k8s.cli.setters.anthos.servicemesh.tag"}
+  tag: 1.8.2-asm.2 # {"$ref":"#/definitions/io.k8s.cli.setters.anthos.servicemesh.tag"}
+  meshConfig:
+    trustDomainAliases: # {"$ref":"#/definitions/io.k8s.cli.setters.anthos.servicemesh.trustDomainAliases"}
+    - "PROJECT_ID.svc.id.goog" # {"$ref":"#/definitions/io.k8s.cli.substitutions.trust-domain"}
+    trustDomain: "PROJECT_ID.svc.id.goog" # {"$ref":"#/definitions/io.k8s.cli.substitutions.trust-domain"}
+    defaultConfig:
+      proxyMetadata:
+        CA_PROVIDER: "GoogleCA"
+        PLUGINS: "GoogleTokenExchange"
+        USE_TOKEN_FOR_CSR: "true"
+        GCE_METADATA_HOST: "metadata.google.internal"
+    # Protocol detection timeout is not supported
+    protocolDetectionTimeout: 0s
+    # Locality load balancing is not supported
+    localityLbSetting:
+      enabled: false
   components:
+    base:
+      enabled: true
+    egressGateways:
+    - enabled: false
+      name: istio-egressgateway
+      k8s:
+        hpaSpec:
+          maxReplicas: 5
+          minReplicas: 2
+    ingressGateways:
+    - enabled: true
+      name: istio-ingressgateway
```

Comment:
it is quite tricky. But the default profile has one more port than the charts only in istio-ingressgateway.

https://github.com/istio/istio/blob/release-1.8/manifests/profiles/default.yaml#L70-L72, which is used for mesh expansion IIRC. We'd better still have them in the yaml:

```

    - name: istio-ingressgateway-181-5
      enabled: true
      k8s:
        env:
          - name: ISTIO_META_ROUTER_MODE
            value: "standard"
        service:
          selector:
            app: istio-ingressgateway
            istio: ingressgateway
            service.istio.io/canonical-revision: asm-181-5
          ports:
            - port: 15021
              targetPort: 15021
              name: status-port
              protocol: TCP
            - port: 80
              targetPort: 8080
              name: http2
              protocol: TCP
            - port: 443
              targetPort: 8443
              name: https
              protocol: TCP
            - port: 15012
              targetPort: 15012
              name: tcp-istiod
              protocol: TCP
            - port: 15443
              targetPort: 15443
              name: tls
              protocol: TCP
```

Reply:
or we can still use asm-gcp profile in 1.8, changing to empty profile is too much for a patch version.

Reply:
this one additional port and ISTIO_META_ROUTER_MODE are inside the charts and only applies to igw with name `istio-ingressgateway`, so won't affect this change.